File: binarySearch.cpp
#include <iostream>
using namespace std;
int binarySearch(int arr[], int size, int key){

    int start = 0;
    int end = size - 1;

    while (start <= end){
        
        int mid = (start + (end-start)/2 );

        if(arr[mid] == key){
        return mid;
        }

        if( key > arr[mid]){
            start = mid+1;
        }
        else{
            end = mid-1;
        }
    }
    cout<<"Element not found"<<endl;
    return -1;
}

int main(){
    int oddArr[5] = {23, 27, 35, 47, 89};
    int evenArr[6] = {12, 15, 17, 49, 67, 89};

    cout<<"The index of 47 in odd waala array is: "<<binarySearch(oddArr, 5, 47)<<endl;
    cout<<"The index of 23 in odd waala array is: "<<binarySearch(oddArr, 5, 23)<<endl;

    cout<<"The index of 89 in even waala array is: "<<binarySearch(evenArr, 6, 89)<<endl;
    cout<<"The index of 15 in even waala array is: "<<binarySearch(evenArr, 6, 15)<<endl;

    cout<<"The index of 100 in even waala array is: "<<binarySearch(evenArr, 6, 100)<<endl;


    return 0;
}

File: binarysearch.py
# Iterative Binary Search Function

def binary_search(arr, x):
	low = 0
	high = len(arr) - 1
	mid = 0

	while low <= high:

		mid = (high + low) // 2

		# If x is greater, ignore left half
		if arr[mid] < x:
			low = mid + 1

		# If x is smaller, ignore right half
		elif arr[mid] > x:
			high = mid - 1

		# means x is present at mid
		else:
			return mid

	# If we reach here, then the element was not present
	return -1


arr=int(input("Enter the array:"))

# Function call
result = binary_search(arr, x)

if result != -1:
	print("Element is present at index", str(result))
else:
	print("Element is not present in array")
